src/miral/CMakeLists.txt

```diff
@@ -27,6 +27,7 @@ add_library(miral-internal STATIC
     xcursor.c                           xcursor.h
                                         join_client_threads.h
                                         window_info_defaults.h
+    window_specification_internal.cpp   window_specification_internal.h
```

Comment:
Why are we splitting an implementation detail of `miral::WindowSpecification` into a separate file? That will just increase build times (and, arguably, make the code harder to follow as it isn't clear that nothing else uses it).

Reply:
I want `make_surface_spec(WindowSpecification const&)` to be a standalone function which isn't in the miral ABI. I get link errors if I put it's implementation in window_specification.cpp, or move window_specification.cpp from the `miral SHARED` library to the `miral-internal STATIC` library in cmake.

Reply:
Repeating back to ensure I understand what you want:

You want `make_surface_spec()` to:

1. have `visibility=hidden` so it isn't exported by libmiral,
2. have access to the implementation of WindowSpecification,
3. not be a member function of WindowSpecification

Isn't it clearer to put it in `window_specification.cpp` and mark it `__attribute__ ((visibility ("hidden")))`?

Reply:
Ah. Further discussion adds another key requirement:

4. available when linking tests against `libmiral-internal`.

That is probably best done by moving all this into `libmiral-internal`.